src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1425,18 +1425,18 @@ private static void ThrowNullLowHighInclusive<T>(T? lowInclusive, T? highInclusi
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe bool SequenceEqual<T>(this Span<T> span, ReadOnlySpan<T> other) where T : IEquatable<T>?
         {
-            int length = span.Length;
-
+            nuint length = (nuint)span.Length;
+            nuint otherLength = (nuint)other.Length;
             if (RuntimeHelpers.IsBitwiseEquatable<T>())
             {
-                return length == other.Length &&
+                return length == otherLength &&
                 SpanHelpers.SequenceEqual(
                     ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(span)),
                     ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(other)),
-                    ((uint)length) * (nuint)sizeof(T));  // If this multiplication overflows, the Span we got overflows the entire address range. There's no happy outcome for this api in such a case so we choose not to take the overhead of checking.
+                    length * (nuint)sizeof(T));  // If this multiplication overflows, the Span we got overflows the entire address range. There's no happy outcome for this api in such a case so we choose not to take the overhead of checking.
```

Comment:
nit: avoid unnecessary promotion in debug build

```suggestion
                    length * (uint)sizeof(T));  // If this multiplication overflows, the Span we got overflows the entire address range. There's no happy outcome for this api in such a case so we choose not to take the overhead of checking.
```